ee/agent/flags/flag_controller.go

```diff
@@ -356,6 +356,46 @@ func (fc *FlagController) OsqueryVerbose() bool {
 	return NewBoolFlagValue(WithDefaultBool(fc.cmdLineOpts.OsqueryVerbose)).get(fc.getControlServerValue(keys.OsqueryVerbose))
 }
 
+func (fc *FlagController) SetEnableWatchdog(enable bool) error {
+	return fc.setControlServerValue(keys.EnableWatchdog, boolToBytes(enable))
+}
+func (fc *FlagController) EnableWatchdog() bool {
+	return NewBoolFlagValue(WithDefaultBool(fc.cmdLineOpts.EnableWatchdog)).get(fc.getControlServerValue(keys.EnableWatchdog))
+}
+
+func (fc *FlagController) SetWatchdogDelaySec(sec int) error {
+	return fc.setControlServerValue(keys.WatchdogDelaySec, intToBytes(sec))
```

Comment:
Is the need to use an int here driven by the osquery options? What I'm getting at is I wonder if we need to introduce another value type to flags or can we make due with existing float and duration.

Reply:
For the delay sec, we could do a duration and convert it to seconds when setting the osquery flag -- but I don't think we should let memory or CPU limit be a float when osquery expects both of those to be integers. It wasn't burdensome to add/maintain the int type and I figure if we don't need it now, we'd want it at some point in the future anyway.